test/jobs/process_entry_job_test.rb

```diff
@@ -75,7 +75,7 @@ class ProcessEntryJobWithFiber < ProcessEntryJob
 
     existing_integrations = Integration.where(tenant: entry.tenant)
     UpdateState.where(model: existing_integrations).delete_all
-    existing_integrations.delete_all
+    existing_integrations.destroy_all
```

Comment:
This is to avoid foreign key constraint violation (together with adding `has_many :integration_states, dependent: :destroy` on the Integration model)